src/anndata/_io/specs/methods.py

```diff
@@ -486,8 +490,14 @@ def write_vlen_string_array(
     dataset_kwargs: Mapping[str, Any] = MappingProxyType({}),
 ):
     """Write methods which underlying library handles nativley."""
-    str_dtype = h5py.special_dtype(vlen=str)
-    f.create_dataset(k, data=elem.astype(str_dtype), dtype=str_dtype, **dataset_kwargs)
+    if elem.shape == ():
+        # extract scalar from array
+        _writer.write_elem(f, k, elem[()], dataset_kwargs=dataset_kwargs)
+    else:
```

Comment:
Can we make this a decorator on these funcitons?

Reply:
Done and done. I wasn't sure where to put the decorator, so it lives in `utils.py` now, let me know if there is a better place.